### PR TITLE
feat: Implement Project.transfer_to

### PR DESCRIPTION
**Pull request by mattrobenolt**
cc @MeredithAnya

### CODE REVIEW

**Review by tkaemming — APPROVED** [review]
Might be reasonable to test slug collisions.

File: src/sentry/models/project.py [hunk 1]
```diff
@@ -348,6 +348,7 @@ def transfer_to(self, team):
         except IntegrityError:
             slugify_instance(self, self.name, organization=organization)
```
Comment (tkaemming) [inline src/sentry/models/project.py]: It's too bad this doesn't return the slug, I'd think it'd be clearer and less error-prone to make this part of the `update` call below. 😞

**Comment by mattrobenolt** [comment]
> Might be reasonable to test slug collisions.

https://github.com/getsentry/sentry/pull/6218/commits/9e32e30faa35b1da8c9d621bb9721afdd9c081fc

**Review by tkaemming — APPROVED** [review]

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**